sympy/polys/eigentools.py

```diff
@@ -0,0 +1,108 @@
+from sympy.polys import Poly
+from sympy.polys.domainmatrix import DDM, ddm_irref, ddm_isub
+from sympy.polys.factortools import dup_factor_list
+from sympy.polys.agca.extensions import FiniteExtension, ExtensionElement
+from sympy.core.symbol import Dummy
+from sympy.polys.rootoftools import CRootOf
+from sympy.polys.polyroots import roots
+
+
+def ddm_nullspace(M):
+    rows, cols = M.shape
+    domain = M.domain
+
+    pivots = ddm_irref(M)
+    rref = M
+
+    basis = []
+    for i in range(cols):
+        if i in pivots:
+            continue
+        vec = [domain.one if i == j else domain.zero for j in range(cols)]
+        for ii, jj in enumerate(pivots):
+            vec[jj] -= rref[ii][i]
+        basis.append(vec)
+
+    return basis
+
+
+def ddm_eigenvects(A, l=Dummy('lambda')):
+    charpoly = A.charpoly()
+    rows, cols = A.shape
+    domain = A.domain
+    _, factors = dup_factor_list(charpoly, domain)
+
+    eigenvects = []
+    for base, exp in factors:
+        if len(base) == 2:
+            field = domain
+            eigenval = -base[1] / base[0]
+
+            AA_items = [[item for item in row] for row in A]
+            AA = DDM(AA_items, (rows, cols), field)
+            EE_items = [
+                [eigenval if i == j else field.zero for j in range(cols)]
+                for i in range(rows)]
```

Comment:
I have attempted to use DDM.eye here, but scalar multiplication is not implemented yet for DDM I think

Reply:
I think it's in #20483